compiler/executable/function.rs

```diff
@@ -40,11 +41,12 @@ pub struct ExecutableFunction {
     // pub plan_cost: f64, // TODO: Where do we fit this in?
 }
 
+#[derive(Debug)]
 pub enum ExecutableReturn {
     Stream(Vec<VariablePosition>),
     Single(SingleSelector, Vec<VariablePosition>),
     Check,
-    Reduce(Vec<ReduceInstruction<VariablePosition>>),
+    Reduce(Arc<ReduceExecutable>),
```

Comment:
They could. I made an Arc because I didn't want to clone the whole `ReduceExecutable`. But we just clone vectors of positions in many places, so I didn't really care about them before. Shall I `Arc` them?